pydrive2/files.py

```diff
@@ -292,6 +336,51 @@ def download(fd, request):
                 if boms:
                     self._RemovePrefix(fd, boms[0])
 
+    @LoadAuth
+    def GetContentIOBuffer(self, mimetype=None, chunksize=DEFAULT_CHUNK_SIZE):
+        """Get a file-like object which has a buffered read() method.
+
+    :param mimetype: mimeType of the file.
+    :type mimetype: str
+    :param chunksize: default read() chunksize.
+    :type chunksize: int
+    :returns: MediaIoReadable -- file-like object.
+    :raises: ApiRequestError, FileNotUploadedError
+    """
+        files = self.auth.service.files()
+        file_id = self.metadata.get("id") or self.get("id")
+        if not file_id:
+            raise FileNotUploadedError()
+
+        # Should use files.export_media instead of files.get_media if
+        # metadata["mimeType"].startswith("application/vnd.google-apps.").
+        # But that would first require a slow call to FetchMetadata().
+        # We prefer to try-except for speed.
+        try:
+            request = files.get_media(fileId=file_id)
+            # Ensures thread safety. Similar to other places where we call
+            # `.execute(http=self.http)` to pass a client from the thread
+            # local storage.
+            if self.http:
+                request.http = self.http
+
+            return MediaIoReadable(request, chunksize=chunksize)
```

Comment:
I think it can be abstracted with a similar block below - lines 377-380

Reply:
yes, this was bugging me too, done now.